### PR TITLE
Feature/awssdk v2 v4 signer support

### LINKED ISSUE #23: client_option - Allow using aws sdk v2 v4 signer
There's WithIAMAuthorization, and only the signer from AWS SDK v1 is supported.
Please add the support for SDK v2 v4 signer: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/aws/signer/v4#pkg-overview
Thanks!

### PR DESCRIPTION
**Pull request by KazuhitoNakamura**
aws SDK v2 v4-signer support (fix #23)

### CODE REVIEW

File: sigv4.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	sdkv2_v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	sdkv1_v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+type sigv4 interface {
+	signHTTP(payload []byte) (http.Header, error)
```
Comment (Naoki-Kato) [inline sigv4.go:20]: It's better to remove argument names or use the same name as the methods

File: sigv4.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			"content-encoding":     req.Header.Get("content-encoding"),
+			"content-type":         req.Header.Get("content-type"),
+			"host":                 req.Host,
+			"x-amz-date":           req.Header.Get("x-amz-date"),
+			"Authorization":        req.Header.Get("Authorization"),
+			"X-Amz-Security-Token": req.Header.Get("X-Amz-Security-Token"),
+		}, nil
+	case *sdkv2_v4.Signer:
+		hash := sha256.Sum256(payload)
+		if err := signer.SignHTTP(context.TODO(), *s.creds, req, hex.EncodeToString(hash[:]), "appsync", s.region, time.Now()); err != nil {
```
Comment (Naoki-Kato) [inline sigv4.go:89]: Please unify the code style with `sdkv1_v4.Signer`
```suggestion
		_, err := signer.SignHTTP(context.TODO(), *s.creds, req, hex.EncodeToString(hash[:]), "appsync", s.region, time.Now())
		if err != nil {
```

Comment (KazuhitoNakamura) [reply sigv4.go:89]: I'm afraid that the SignHTTP only returns an error value, so I do not think there is a problem.
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2@v1.18.1/aws/signer/v4#Signer.SignHTTP

Comment (Naoki-Kato) [reply sigv4.go:89]: I got it, sorry for my mistake.

File: sigv4.go (hunk 1 above)
Comment (KazuhitoNakamura) [reply sigv4.go:20]: Sorry, I don't get your intent of "use the same name as the method".

For example, cloud
```
signHTTP(http []byte) (http.Header, error)
```
meet your expectations?

Comment (Naoki-Kato) [reply sigv4.go:20]: I'd just like to ask difference in use between `body []byte` of the implemented method and `payload []byte` of interface.

Comment (KazuhitoNakamura) [reply sigv4.go:20]: Thanks, I understand.
I will update it soon.

**Comment by KazuhitoNakamura** [comment]
Hello @Naoki-Kato, thank you for your comments.
I have corrected the points you pointed out.
Could you check it again?

**Review by Naoki-Kato — APPROVED** [review]
LGTM!

**Pull request merged**